via/views/api/exceptions.py

```diff
@@ -1,11 +1,46 @@
-from pyramid.view import forbidden_view_config, notfound_view_config, view_defaults
+import logging
+
+from h_pyramid_sentry import report_exception as report_exception_to_sentry
+from pyramid.view import (
+    exception_view_config,
+    forbidden_view_config,
+    notfound_view_config,
+    view_defaults,
+)
+
+logger = logging.getLogger(__name__)
 
 
 @view_defaults(path_info="^/api/.*", renderer="json")
 class APIExceptionViews:
-    def __init__(self, request):
+    def __init__(self, context, request):
+        self.context = context
         self.request = request
 
+    @exception_view_config(Exception)
+    def exception(self):
+        report_exception_to_sentry(self.context)
+        logger.exception(self.context)
+
+        self.request.response.status_int = 500
+
+        return {
+            "errors": [
+                {
+                    "status": self.request.response.status_int,
+                    "code": self.context.__class__.__name__,
+                    "title": str(
+                        getattr(
+                            self.context,
+                            "cause",
+                            "Something went wrong",
```

Comment:
The generic fallback error message [used to be](https://github.com/hypothesis/via/blob/bd1241cab4bbef9ef222c5f82fe1270e2b0cd02d/via/views/api/youtube.py#L36) view-specific (`"Failed to get transcript from YouTube"`) but now it has to be fully generic.

There might be a way for each view to specify a fallback error message for the exception to use but I think we want to avoid every view having to catch and re-raise exceptions as that would lose part of the benefit of exception views (that views and their tests don't need to bother with exception handling). Perhaps something in the `@view_config`?